runtime/runtime/src/ext.rs

```diff
@@ -158,13 +177,13 @@ impl<'a> External for RuntimeExt<'a> {
             }
             Some(key) => {
                 let key = key.map_err(wrap_error)?;
-                Ok(Some((
-                    key[self.storage_prefix.len()..].to_vec(),
-                    self.trie_update
-                        .get(&key)
-                        .expect("error cannot happen")
-                        .expect("key is guaranteed to be there"),
-                )))
+                let ptr = self
+                    .trie_update
+                    .get_ref(&key)
+                    .expect("error cannot happen")
+                    .expect("key is guaranteed to be there");
+                let result = Box::new(RuntimeExtValuePtr { ptr }) as Box<_>;
```

Comment:
Once we implement `ValuePtr` for `ValuePointer` we can simple return `here` without wrapping it into a box, and use `&'b dyn ValuePtr + 'b` in the return type.

Reply:
ValuePointer is still a local variable, we can't return a reference to it.

Reply:
If we got rid of `RuntimeExtValuePtr`, `ptr` would've been `&dyn ValuePtr` because of the semantics of `get_ref` method. `ptr` is a local variable, but we are not returning a reference to it, but we are returning it directly.

However, I understand the motivation behind having `RuntimeExtValuePtr` so it is irrelevant now.

Reply:
`TrieUpdate::get_ref` creates and returns the object - there is no one who can own it